src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs

```diff
@@ -70,7 +70,7 @@ internal override DiagnosticInfo ErrorInfo
                 // Since we do not know what task was being performed, for now we just report a generic
                 // "you must add a reference" error.
 
-                if (containingAssembly.IsMissing)
+                if (containingAssembly?.IsMissing == true)
```

Comment:
Did you consider marking `this.ContainingAssembly` as having a `AssemblySymbol?` return?

Reply:
No. It is defined in a file that is not annotated. The problem wasn't that we didn't know that the ContainingAssembly can return null in general. The problem was that we assumed that that cannot happen for this specific flavor of symbol.